### PR TITLE
Merge values in authtype and record auth details in AuthMethod

### PR DESCRIPTION
**Pull request by Flanker32**
Merge properties in `AuthType` and add `AuthMethod` to reduce connection between configuration and implementation.

### CODE REVIEW

File: azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java [hunk 1]
```diff
@@ -37,8 +38,7 @@
 
 public class AzureAuthHelper {
 
-    private static final AuthType[] AUTH_ORDER = {AuthType.SERVICE_PRINCIPAL, AuthType.AZURE_SECRET_FILE, AuthType.CLOUD_SHELL,
-        AuthType.AZURE_CLI, AuthType.OAUTH, AuthType.DEVICE_LOGIN};
+    private static final AuthType[] AUTH_ORDER = {AuthType.SERVICE_PRINCIPAL, AuthType.AZURE_SECRET_FILE, AuthType.AZURE_CLI, AuthType.AZURE_MAVEN_PLUGIN};
```
Comment (andxu) [inline azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java]: azure_maven_login

Comment (Flanker32) [reply azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java]: Changed to `azure_auth_maven_plugin`

File: azure-auth-helper/src/main/java/com/microsoft/azure/auth/configuration/AuthMethod.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * license information.
+ */
+
+package com.microsoft.azure.auth.configuration;
+
+/**
+ * Enum records detail auth method
+ */
+public enum AuthMethod {
+    AZURE_CLI,
```
Comment (andxu) [inline azure-auth-helper/src/main/java/com/microsoft/azure/auth/configuration/AuthMethod.java]: where is azure auth plugin?

Comment (Flanker32) [reply azure-auth-helper/src/main/java/com/microsoft/azure/auth/configuration/AuthMethod.java]: Divided into 3 parts:
- AZURE_SECRET_FILE
- OAUTH
- DEVICE_LOGIN

**Pull request merged**